doc/src/manual/calling-c-and-fortran-code.md

```diff
@@ -763,11 +770,13 @@ that has no internal fields and whose sole purpose is to be placed in the type p
 `Ptr` type.  The return type of the [`ccall`](@ref) is declared as `Ptr{gsl_permutation}`, since
 the memory allocated and pointed to by `output_ptr` is controlled by C (and not Julia).
 
-The input `n` is passed by value, and so the function's input signature is simply declared as
-`(Csize_t,)` without any `Ref` or `Ptr` necessary. (If the wrapper was calling a Fortran function
-instead, the corresponding function input signature should instead be `(Ref{Csize_t},)`, since
-Fortran variables are passed by reference.) Furthermore, `n` can be any type that is convertable
-to a `Csize_t` integer; the [`ccall`](@ref) implicitly calls [`Base.cconvert(Csize_t, n)`](@ref).
+The input `n` is passed by value, and so the function's input signature is
+simply declared as `(Csize_t,)` without any `Ref` or `Ptr` necessary. (If the
+wrapper was calling a Fortran function instead, the corresponding function input
+signature should instead be `(Ref{Csize_t},)`, since Fortran variables are
+passed by pointers.) Furthermore, `n` can be any type that is convertable to a
```

Comment:
This should still say "by reference", since we're referring to Fortran here.

"GNU Fortran passes most arguments by reference, i.e. by passing a pointer to the data" (https://gcc.gnu.org/onlinedocs/gfortran/Argument-passing-conventions.html)

Reply:
It should not since there's a different and conflicting meaning of the term even though Fortran did use it in a different way first.

Reply:
IIRC, Fortran actually does pass arguments by reference though, since it doesn't have pointers and the variables themselves are values, not bindings.

Reply:
Yes, but nothing from Julia's perspective is "passed by reference" – from Julia's perspective, mutable objects are passed by sharing to a function which cannot change any bindings in the caller but can mutate the shared objects.

Reply:
Yes, that's fine. But the text above is literally "... Fortran variables are passed by reference". We can reword it to clarify that the subject is Julia "..., since Fortran expects Julia to pass a pointer to the value." But in the above clause, the subject is Fortran. Another way to say it is that "..., since Fortran passes variables by reference, which is equivalent in Julia to passing a pointer"

Reply:
That would be fine but saying that they are "passed by pointers" is also correct.